lib/modules/federation_identity_lookup/domain/models/federation_third_party_contact.dart

```diff
@@ -0,0 +1,145 @@
+import 'dart:convert';
+
+import 'package:crypto/crypto.dart';
+import 'package:equatable/equatable.dart';
+import 'package:fluffychat/modules/federation_identity_lookup/domain/models/federation_hash_details_response.dart';
+import 'package:fluffychat/utils/string_extension.dart';
+import 'package:matrix/encryption/utils/base64_unpadded.dart';
+
+abstract class FederationThirdPartyContact with EquatableMixin {
+  final String? matrixId;
+
+  final String thirdPartyIdType;
+
+  final String thirdPartyId;
+
+  final Map<String, String>? thirdPartyIdToHashMap;
+
+  FederationThirdPartyContact({
+    this.matrixId,
+    required this.thirdPartyIdType,
+    required this.thirdPartyId,
+    this.thirdPartyIdToHashMap,
+  });
+
+  @override
+  List<Object?> get props => [
+        matrixId,
+        thirdPartyIdType,
+        thirdPartyId,
+        thirdPartyIdToHashMap,
+      ];
+
+  String calculateHashWithAlgorithmSha256({
+    required String pepper,
+  }) {
+    final input = [thirdPartyId, thirdPartyIdType, pepper].join(' ');
+    final bytes = utf8.encode(input);
+    final lookupHash =
+        encodeBase64Unpadded(sha256.convert(bytes).bytes).urlSafeBase64;
+    return lookupHash;
+  }
+
+  String calculateHashWithoutAlgorithm({
+    required String pepper,
+  }) {
+    return [thirdPartyId, thirdPartyIdType].join(' ');
+  }
+
+  Set<String> calculateHashUsingAllPeppers({
```

Comment:
Verify the number of hashes for all cases:
- multiple algorithms: `none` or `sha256`
- multiple peppers